File: test/samples/lists/lists.rs
/// * `n` - the first list's size
/// * `list_int` - a list containing ints
/// * `size` - an other size
/// * `list_char` - a list of char
/// * `string` - a string
/// * `list_string4` - a list of strings of size 4
/// * `list_list_string2` - a list of list of strings of size 2 of size 2 of size 2
/// * `matrix` - a matrix of int
fn lists(n: i32, list_int: Vec<i32>, size: i32, list_char: Vec<char>, string: String, list_string4: Vec<String>, list_list_string2: Vec<Vec<String>>, matrix: Vec<Vec<i32>>) {
    /* TODO Aren't these lists beautifull? */
}

fn main() {
    let mut buffer = String::new();

    let n = read_line(&mut buffer)
        .parse()
        .expect("invalid `N` parameter");

    let list_int = read_line(&mut buffer)
        .split_whitespace()
        .map(str::parse)
        .collect::<Result<_, _>>()
        .expect("invalid `list int` parameter");

    let size = read_line(&mut buffer)
        .parse()
        .expect("invalid `size` parameter");

    let list_char = read_line(&mut buffer).chars().collect();

    let string = read_line(&mut buffer).to_string();

    let list_string4 = (0..size)
        .map(|_| read_line(&mut buffer).to_string())
        .collect();

    let list_list_string2 = (0..2)
        .map(|_| {
            (0..2)
                .map(|_| read_line(&mut buffer).to_string())
                .collect()
        })
        .collect();

    let matrix = (0..size)
        .map(|_| {
            read_line(&mut buffer)
                .split_whitespace()
                .map(str::parse)
                .collect::<Result<_, _>>()
        })
        .collect::<Result<_, _>>()
        .expect("invalid `matrix` parameter");

    lists(n, list_int, size, list_char, string, list_string4, list_list_string2, matrix);
}

fn read_line(mut buffer: &mut String) -> &str {
    buffer.clear();
    std::io::stdin()
        .read_line(&mut buffer)
        .expect("impossible to read a new line");
    buffer.trim_end()
}

File: test/samples/lists/lists.lua
-- n (number): the first list's size
-- list_int (table: array[number]): a list containing ints
-- size (number): an other size
-- list_char (table: array[string]): a list of char
-- string_ (string): a string
-- list_string4 (table: array[string]): a list of strings of size 4
-- list_list_string2 (table: array[array[string]]): a list of list of strings of size 2 of size 2 of size 2
-- matrix (table: array[array[number]]): a matrix of int
function lists(n, list_int, size, list_char, string_, list_string4, list_list_string2, matrix)
    -- TODO Aren't these lists beautifull?
end

local n = tonumber(io.read())
local list_int = {}
for i in string.gmatch(io.read(), "-?%d+") do
    table.insert(list_int, tonumber(i))
end
local size = tonumber(io.read())
local list_char = {}
io.read():gsub(".",function(i) table.insert(list_char, i) end)
local string_ = io.read()
local list_string4 = {}
for i = 1, size do
    list_string4[i] = io.read()
end
local list_list_string2 = {}
for i = 1, 2 do
    list_list_string2[i] = {}
    for j = 1, 2 do
        list_list_string2[i][j] = io.read()
    end
end
local matrix = {}
for i = 1, size do
    matrix[i] = {}
    for j in string.gmatch(io.read(), "-?%d+") do
        table.insert(matrix[i], tonumber(j))
    end
end

lists(n, list_int, size, list_char, string_, list_string4, list_list_string2, matrix)
